sympy/polys/multivariate_resultants.py

```diff
@@ -185,6 +185,58 @@ def get_dixon_matrix(self, polynomial):
 
         return dixon_matrix
 
+    def KSY_precondition(self, matrix):
+        """
+        Test for the validity of the Kapur-Saxena-Yang precondition.
+
+        The precondition requires that the column corresponding to the
+        monomial 1 = x_1 ^ 0 * x_2 ^ 0 * ... * x_n ^ 0 is not a linear
+        combination of the remaining ones. In sympy notation this is
+        the last column. For the precondition to hold the last non-zero
+        row of the rref matrix should be of the form [0, 0, ..., 1].
+        """
+        if matrix.is_zero:
+            return False
+
+        m, n = matrix.shape
+
+        # simplify the matrix and keep only its non-zero rows
+        matrix = simplify(matrix.rref()[0])
+        rows = [i for i in range(m) if any(matrix[i, j] != 0 for j in range(n))]
+        matrix = matrix[rows,:]
+
+        condition = Matrix([[0 for i in range(n-1)] + [1]])
```

Comment:
I think the argument could be `[0]*(n-1) + [1]`.

Reply:
Thank you. This looks better.